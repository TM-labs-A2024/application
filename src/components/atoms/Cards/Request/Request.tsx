import { Link } from '@chakra-ui/next-js'
import { Text } from '@chakra-ui/react'
import NextLink from 'next/link'
import React from 'react'

import Chevron from '../../../../../public/static/icons/chevron.svg'

export default function Request({
  href,
  title,
  description
}: {
  href: string
  title: string
  description: string
}) {
  return (
    <Link as={NextLink} href={href} className="w-full" data-testid="request-card">
      <div className="flex flex-row items-center justify-between">
        <div className="flex flex-col items-start gap-1">
          <Text fontSize="md" as="b">
            {title}
          </Text>
          <Text>{description}</Text>
        </div>
        <Chevron />
      </div>
    </Link>
  )
}
