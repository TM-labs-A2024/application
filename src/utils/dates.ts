export const formatDate = (date: string) => {
  const chunks = date.split('-')

  const newDate = chunks[1] + '-' + chunks[2] + '-' + chunks[0]

  return newDate
}
