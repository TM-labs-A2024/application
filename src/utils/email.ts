import emailjs from '@emailjs/browser'

export function sendEmail(data: {
  from_name: string
  to_name: string
  code: string | number
  to_email: string
}) {
  const templateParams = {
    ...data
  }

  const serviceKey = process.env.NEXT_PUBLIC_EMAIL_SERVICE_KEY ?? ''
  const templateKey = process.env.NEXT_PUBLIC_EMAIL_TEMPLATE_KEY ?? ''
  const apiKey = process.env.NEXT_PUBLIC_EMAIL_API_KEY ?? ''

  emailjs.send(serviceKey, templateKey, templateParams, apiKey).catch(() => {
    alert('Algo sali√≥ mal, por favor reintenta el registro.')
  })
}
