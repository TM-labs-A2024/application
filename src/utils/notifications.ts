import { GENERIC_ERROR } from '@constants/index'
import { isMobile } from '@utils/index'
import { Store } from 'react-notifications-component'

export const setupErrorNotification = () => {
  Store.addNotification(
    GENERIC_ERROR('Algo salió mal, por favor inténtelo de nuevo.', isMobile(window))
  )
}
