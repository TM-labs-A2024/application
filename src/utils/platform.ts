import { Capacitor } from '@capacitor/core'

export const isWeb = () => Capacitor.getPlatform() === 'web'

export const isIOS = () => Capacitor.getPlatform() === 'ios'

export const isAndroid = () => Capacitor.getPlatform() === 'android'
