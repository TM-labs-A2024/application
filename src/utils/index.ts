import { formatDate } from './dates'
import { sendEmail, generateVerificationCode, setupEmailSending } from './email'
import { setupErrorNotification } from './notifications'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export {
  isWeb,
  isIOS,
  isAndroid,
  isMobile,
  sendEmail,
  generateVerificationCode,
  setupEmailSending,
  setupErrorNotification,
  formatDate
}
