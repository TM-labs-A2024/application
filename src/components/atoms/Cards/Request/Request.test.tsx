import { render, screen } from '@test/utils/index'

import '@testing-library/jest-dom'
import Request from './Request'

const request = {
  href: '/institucion/medico/1000000',
  title: 'José Pérez',
  description: 'C.I: 1000000, Alergología.'
}

describe('Atoms > Cards > Request test', () => {
  test('The card shows the title and description of the request', async () => {
    render(<Request {...request} />)

    expect(await screen.findByText(request.title)).toBeInTheDocument()
    expect(await screen.findByText(request.description)).toBeInTheDocument()
  })

  test('The url matches the expected', async () => {
    render(<Request {...request} />)

    const enlace = await screen.findByRole('link')
    expect(enlace).toHaveAttribute('href', request.href)
  })

  test('Matches the snapshot', async () => {
    render(<Request {...request} />)

    expect(await screen.findByTestId('request-card')).toMatchSnapshot()
  })
})
