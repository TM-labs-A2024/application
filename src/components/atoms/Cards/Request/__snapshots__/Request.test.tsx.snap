// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms > Cards > Request test Matches the snapshot 1`] = `
<a
  class="chakra-link w-full css-spn4bz"
  data-testid="request-card"
  href="/institucion/medico/1000000"
>
  <div
    class="flex flex-row items-center justify-between"
  >
    <div
      class="flex flex-col items-start gap-1"
    >
      <b
        class="chakra-text css-qqfgvy"
      >
        José Pérez
      </b>
      <p
        class="chakra-text css-0"
      >
        C.I: 1000000, Alergología.
      </p>
    </div>
    <div />
  </div>
</a>
`
